attestation-service/attestation-service/src/policy_engine/opa/mod.rs

```diff
@@ -88,82 +81,56 @@ impl PolicyEngine for OPA {
                 hex::encode(hex)
             };
 
-            let policy_go = GoString {
-                p: policy.as_ptr() as *const c_char,
-                n: policy.len() as isize,
-            };
-
-            let reference = serde_json::json!({ "reference": reference_data_map }).to_string();
-
-            let reference_go = GoString {
-                p: reference.as_ptr() as *const c_char,
-                n: reference.len() as isize,
-            };
-
-            let input_go = GoString {
-                p: input.as_ptr() as *const c_char,
-                n: input.len() as isize,
-            };
-
-            // Call the function exported by cgo and process the returned decision
-            let decision_buf: *mut c_char =
-                unsafe { evaluateGo(policy_go, reference_go, input_go) };
-            let decision_str: &CStr = unsafe { CStr::from_ptr(decision_buf) };
-            let policy_res = decision_str.to_str()?.to_string();
-            debug!("OPA Evaluated: {}", policy_res);
-            if policy_res.starts_with("Error::") {
-                bail!("OPA verification failed: {policy_res}");
-            }
+            // Add policy as data
+            engine
```

Comment:
Since `policy` has already been read from file, you could use 
[Engine::add_policy](https://docs.rs/regorus/0.1.5/regorus/struct.Engine.html#method.add_policy) to avoid a reread from disk.

Reply:
Does the first parameter `path` result in a fs reading?

Reply:
No. It is intended to be the path/name of the file from which the policy string came, but you can pass in any string.
It is mainly used in error messages to indicate which policy file the error originated from. In the example below, `mypolicy.rego` was passed in as the path.
```
Error: 
--> mypolicy.rego:3:11
  |
3 | x = count1(input.data)
  |           ^
error: could not find function count1
```

Reply:
Thanks!